pkg/reconciler/v1alpha1/configuration/configuration.go

```diff
@@ -228,3 +249,27 @@ func (c *Reconciler) updateStatus(u *v1alpha1.Configuration) (*v1alpha1.Configur
 	}
 	return newu, nil
 }
+
+func (c *Reconciler) gcRevisions(ctx context.Context, config *v1alpha1.Configuration) error {
+	logger := logging.FromContext(ctx)
+
+	selector := labels.Set{serving.ConfigurationLabelKey: config.Name}.AsSelector()
+	revs, err := c.revisionLister.Revisions(config.Namespace).List(selector)
+	if err != nil {
+		return err
+	}
+
+	for _, rev := range revs {
+		isStale, err := resources.IsRevisionStale(rev, c.clock)
```

Comment:
We also need to check whether this Revision is `Latest{Created,Ready}RevisionName` before deleting it.